File: app/src/main/java/com/di/penopllast/vklikesremover/di/ComponentsHolder.java
package com.di.penopllast.vklikesremover.di;

import com.di.penopllast.vklikesremover.application.DislikerApp;
public class ComponentsHolder {

    private final DislikerApp context;
    private AppComponent appComponent;

    public ComponentsHolder(DislikerApp context) {
        this.context = context;
    }

    public void init() {
        appComponent = DaggerAppComponent
                .builder()
                .application(context)
                .build();
    }

    public AppComponent getAppComponent() {
        return appComponent;
    }

}

File: app/src/main/java/com/di/penopllast/vklikesremover/di/ComponentsHolder.kt
package com.di.penopllast.vklikesremover.di

import com.di.penopllast.vklikesremover.application.DislikerApp

class ComponentsHolder(private val context: DislikerApp) {
    var appComponent: AppComponent? = null
        private set

    fun init() {
        appComponent = DaggerAppComponent
                .builder()
                .application(context)
                .build()
    }

}
